fbrp/src/fbrp/runtime/conda.py

```diff
@@ -174,107 +172,187 @@ async def handle_down(self):
 
 
 class Conda(BaseRuntime):
+    class SharedEnv:
+        def __init__(
+            self,
+            name,
+            use_named_env=None,
+            copy_named_env=None,
+            yaml=None,
+            channels=[],
+            dependencies=[],
+            use_mamba=None,
+        ):
+            if use_named_env and any([copy_named_env, yaml, channels, dependencies]):
+                raise ValueError(
+                    f"'use_named_env' cannot be mixed with other Conda environment commands."
+                )
+
+            self.name = name
+            self.use_named_env = use_named_env
+            self.copy_named_env = copy_named_env
+            self.yaml = yaml
+            self.conda_env = CondaEnv(channels[:], dependencies[:])
+            self.use_mamba = (
+                use_mamba if use_mamba is not None else bool(shutil.which("mamba"))
+            )
+            self._built = False
+
+            if use_named_env:
+                self._validate_use_named_env()
+
+        def _validate_use_named_env(self):
+            result = subprocess.run(
+                f"""
+                    eval "$(conda shell.bash hook)"
+                    conda activate {self.use_named_env}
+                """,
+                shell=True,
+                executable="/bin/bash",
+                stderr=subprocess.PIPE,
+            )
+            if result.returncode:
+                raise RuntimeError(f"'use_named_env' not valid: {result.stderr}")
+
+        def _generate_env_content(self, root: pathlib.Path):
```

Comment:
It's not entirely clear at first glance what this method does -- mind adding a docstring?

Reply:
Done